### PR TITLE
[V0184] Make endpoint /presidential/financial_summary/

### PR DESCRIPTION
**Pull request by lbeaufort**
## Summary (required)

- Partial resolution for #4178 

 Make endpoint `/presidential/financial_summary/`

## How to test the changes locally

- `export FEC_FEATURE_PRESIDENTIAL=True`
- Change `PresidentialSummary` tablename to `'ofec_presidential_financial_summary_vw_lb'`
- Test a query with Swagger

**All:** 
- http://localhost:5000/v1/presidential/financial_summary

**Candidate ID:** 
- http://localhost:5000/v1/presidential/financial_summary/?candidate_id=P00000001
- http://localhost:5000/v1/presidential/financial_summary/?candidate_id=P00009621

**Election_year:**
- http://localhost:5000/v1/presidential/financial_summary/?election_year=2020
- Test migration with 
```
dropdb cfdm_test
createdb cfdm_test
invoke create_sample_db
flyway migrate
```
## Impacted areas of the application
List general components of the application that this PR will affect:

- Presidential map data, right-hand chart  "Summary" and "Expenditures"

Screen Shot 2020-02-06 at 2 32 02 PM

Screen Shot 2020-02-06 at 2 31 44 PM

## Related PRs
https://github.com/fecgov/openFEC/pull/4190
https://github.com/fecgov/openFEC/pull/4181

### CODE REVIEW

**Comment by fec-jli** [comment]
Great. this sql is a hard work. @lbeaufort 
just think about, two cents here:
1) if we can select one or two more columns from this query, it can feed the data for **left side candidate list** on national map page.  
2) same as other endpoint, need union multi-election_year and add one more column: election_year. Thanks

**Comment by lbeaufort** [comment]
@fec-jli thanks for taking a look at my WIP PR!
>if we can select one or two more columns from this query, it can feed the data for left side candidate list on national map page.

I will add them to SQL so if we make the change it can just be on the python change

>same as other endpoint, need union multi-election_year and add one more column: election_year. Thanks
😁 thank you! Definitely hadn't caught that yet.

File: tests/test_presidential.py [hunk 1]
```diff
@@ -137,3 +138,57 @@ def test_sort(self):
...
+        for field, example in filter_fields:
+            page = api.url_for(PresidentialSummaryView, **{field: example})
+            # returns at least one result
+            results = self._results(page)
+            self.assertGreater(len(results), 0)
+            # doesn't return all results
+            response = self._response(page)
+            print("field=" + field)
+            print("original_count=" + str(original_count))
+            print("count=" + str(response['pagination']['count']))
```
Comment (fec-jli) [inline tests/test_presidential.py]: Can you remove these print ? Thanks

**Comment by fec-jli** [comment]
Great job. work as expected. I will leave Jean to review sql file.
one minor comment to clean the debug print lines in test_presidential.py.
Thanks

**Comment by lbeaufort** [comment]
@fec-jli good catch, thanks! I pushed a commit: https://github.com/fecgov/openFEC/pull/4189/commits/77445751a343bac58878a0a76cbc4c79fe1969cf

**Comment by fecjjeng** [comment]
In general, specify the columns you want instead of * is a better practice.  But in this particular case, since we are very sure these two tables are of exactly the same structure, not listing all the columns (which is a lot) improve the readability.  Just be cautious while doing so.

**Pull request merged**